pulsar-metadata/src/main/java/org/apache/pulsar/metadata/cache/impl/MetadataCacheImpl.java

```diff
@@ -229,7 +240,11 @@ public MetadataCacheImpl(MetadataStore store, MetadataSerde<T> serde) {
         return store.delete(path, Optional.empty())
                 .thenAccept(v -> {
                     // Mark in the cache that the object was removed
-                    objCache.put(path, FutureUtils.value(Optional.empty()));
+                    // objCache.put(path, FutureUtils.value(Optional.empty()));
```

Comment:
I think that we could either invalidate it from the cache here, do the read to set the watch, or again, since it's mostly related with the ZK, do it there

Reply:
I take this back, the cache is anyway invalidated in `AbstractMetadataStore.delete()`. We can just remove the commented line here.

Reply:
Yes the cache gets invalidated on delete. I will remove the commented line